src/linux/backend-v4l2.cpp

```diff
@@ -546,6 +546,65 @@ namespace librealsense
             }
         }
 
+        std::string get_devname_from_video_path(const std::string& video_path)
+        {
+            std::string dev_name;
+
+            std::ifstream uevent_file(video_path + "/uevent");
+            if (!uevent_file)
+            {
+                LOG_INFO("Cannot access " + video_path + "/uevent");
+                return dev_name;
+            }
+
+            std::string uevent_line, major_string, minor_string;
+            while (std::getline(uevent_file, uevent_line) && (major_string.empty() || minor_string.empty()))
+            {
+                if (uevent_line.find("MAJOR=") != std::string::npos)
+                {
+                    major_string = uevent_line.substr(uevent_line.find_last_of('=') + 1);
+                }
+                else if (uevent_line.find("MINOR=") != std::string::npos)
+                {
+                    minor_string = uevent_line.substr(uevent_line.find_last_of('=') + 1);
+                }
+            }
+            uevent_file.close();
+
+            if (major_string.empty() || minor_string.empty())
+            {
+                LOG_INFO("No Major or Minor number found for " + video_path);
+                return dev_name;
+            }
+
+            DIR * dir = opendir("/dev");
+            if (!dir)
+            {
+                LOG_INFO("Cannot access /dev");
```

Comment:
Isn't this a LOG_ERROR?

Reply:
Maybe other new logs are error/warning too?

Reply:
+ lets add errno string to the log

Reply:
> Isn't this a LOG_ERROR?

Not being able to access `/dev` is almost certainly an error, but the others are not.

In cases where more aspects of the host's filesystem and devices are hidden from the container's perspective, disagreements between the sysfs and devfs are ignored. Only when a video device in sysfs is also present in the devfs, is that node added to the video_paths vector. Even if none are found it's not an error and shouldn't be fatal, in the case for example where only MIPI devices are passed into the container.

Reply:
There are potential TOCTOU issues with the `get_devname_from_video_path` function at the moment on lines 707 and 963. Will refactor the function to take the string to be returned as an argument by reference, and return an error code that can be checked.

Reply:
> Isn't this a LOG_ERROR?

Actually the original reason I used `LOG_INFO` over `LOG_ERROR` here was that the same approach was previously taken on line 612 when accessing `/sys/class/video4linux`.